### DISCUSSION TITLE
CheckOptions method to check multiple options identified by the selector

### DISCUSSION THREAD

**Discussion by cf-pkeating** [Codeception/Codeception#6175] [Ideas]
While putting together a performance test, I wanted there to be a method that checks options by class name. I created this method in InnerBrowser.php, which works for our particular case. I thought it might provide a base for a future addition. I make no claims for its quality. It is a basic hack. Let me know whether you think this would be a useful addition.

``` php
    /**
    * Checks all nodes matched by the selector. 
    *
    * @param $selector
    * @return void
    */
    public function checkOptions($selector)
    {
        $nodes = $this->match($selector);

        foreach($nodes as $nodeIndex => $node){
            if($node->hasAttribute('id')){
                $nodeId = $node->getAttribute('id');
                $this->checkOption('#'.$nodeId);
            }
            elseif($node->hasAttribute('name')){
                $nodeName = $node->getAttribute('name');
                $this->checkOption($nodeName);
            }       
        }
    }
```

**Comment by zbateson** [comment]
Hmmm... does `$I->checkOption('.className');` not work for some reason? It should...

**Comment by cf-pkeating** [comment]
When I ran the test in debug mode using checkOption, the POST data only contained one item that had been checked. The ChoiceFormField returned by proceedCheckOption appears to match a single form field, i.e. the first one. This item is then ticked.

``` php

    public function checkOption($option)
    {
        $this->proceedCheckOption($option)->tick();
    }

     /**
     * @param $option
     * @return ChoiceFormField
     */
    protected function proceedCheckOption($option)
    {
        $form = $this->getFormFor($field = $this->getFieldByLabelOrCss($option));
        $name = $field->attr('name');

        if ($field->getNode(0) === null) {
            throw new TestRuntime("Form field $name is not located");
        }
        // If the name is an array than we compare objects to find right checkbox
        $formField = $this->matchFormField($name, $form, new ChoiceFormField($field->getNode(0)));
        if (!$formField instanceof ChoiceFormField) {
            throw new TestRuntime("Form field $name is not a checkable");
        }
        return $formField;
    }

```

**Comment by zbateson** [comment]
Aaah, I see what you're saying.  I'll consider improvements to proceedCheckOption on that basis... perhaps instead of simply checking off one, it could return and check "all" options returned by the selector.  I'm not sure I like the idea of introducing another function for that at this point simply because the class is getting a bit overloaded and I'm hoping to simplify.

If you have some time, please consider submitting a pull request against fixing proceedCheckOption.  I must admit I'm not entirely certain without spending some time if such a change would break some expectations... but I can't think of any off-hand.  Preferably the same change would be applied to WebDriver as well.

Thanks for the report :+1:

**Comment by cf-pkeating** [comment]
Sorry for the lateness of my reply. Have been a bit busy lately. Submitted a pull request this morning.

**Comment by Naktibalda** [comment]
Pull request  #2166 was not merged.
